packages/ibm-products/src/components/TruncatedList/TruncatedList.test.js

```diff
@@ -0,0 +1,122 @@
+import { ListItem } from '@carbon/react';
+
+const blockClass = `${pkg.prefix}--truncated-list`;
+const componentName = TruncatedList.displayName;
+
+// values to use
+const className = `class-${uuidv4()}`;
+const dataTestId = uuidv4();
+
+describe(componentName, () => {
+  const { getComputedStyle } = window;
+
+  beforeEach(() => {
+    window.getComputedStyle = jest.fn();
+  });
+
+  afterEach(() => {
+    window.getComputedStyle = getComputedStyle;
+  });
+
+  it('renders a component TruncatedList', async () => {
+    render(
+      <TruncatedList>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+    expect(document.querySelector(`.${blockClass}`)).toBeInTheDocument();
+  });
+
+  it('has no accessibility violations', async () => {
+    const { container } = render(
+      <TruncatedList>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+    expect(container).toBeAccessible(componentName);
+    expect(container).toHaveNoAxeViolations();
+  });
+
+  it(`renders children`, async () => {
+    render(
+      <TruncatedList>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+    screen.getByText(/Item 2/);
+  });
+
+  it('applies className to the containing node', async () => {
+    render(
+      <TruncatedList className={className}>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+    expect(document.querySelector(`.${blockClass}`)).toHaveClass(className);
+  });
+
+  it('adds additional props to the containing node', async () => {
+    render(
+      <TruncatedList data-testid={dataTestId}>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+    screen.getByTestId(dataTestId);
+  });
+
+  it('forwards a ref to an appropriate node', async () => {
+    const ref = React.createRef();
+    render(
+      <TruncatedList ref={ref}>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+    expect(ref.current).toHaveClass(blockClass);
+  });
+
+  it('adds the Devtools attribute to the containing node', async () => {
+    render(
+      <TruncatedList data-testid={dataTestId}>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+      </TruncatedList>
+    );
+
+    expect(screen.getByTestId(dataTestId)).toHaveDevtoolsAttribute(
+      componentName
+    );
+  });
+
+  it('renders a limited number of items when collapsed', async () => {
+    render(
+      <TruncatedList collapsedItemsLimit={2} expandedItemsLimit={4}>
+        <ListItem>Item 1</ListItem>
+        <ListItem>Item 2</ListItem>
+        <ListItem>Item 3</ListItem>
+        <ListItem>Item 4</ListItem>
+        <ListItem>Item 5</ListItem>
+      </TruncatedList>
+    );
+    expect(document.querySelectorAll(`li`).length).toBe(2);
```

Comment:
Use `getByRole` here to avoid using `document.querySelector`.

Possibly something like the follow:
```js
expect(screen.getByRole('listitem').length).toBe(2);
```